daemon/access_test.go

```diff
@@ -0,0 +1,186 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package daemon
+
+import (
+	"bytes"
+	"net/http/httptest"
+
+	. "gopkg.in/check.v1"
+
+	"github.com/snapcore/snapd/client"
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/logger"
+	"github.com/snapcore/snapd/overlord/auth"
+	"github.com/snapcore/snapd/polkit"
+	"github.com/snapcore/snapd/testutil"
+)
+
+type accessSuite struct{}
+
+var _ = Suite(&accessSuite{})
+
+func (s *accessSuite) TestOpenAccess(c *C) {
+	var ac accessChecker = openAccess{}
+
+	// openAccess denies access from snapd-snap.socket
+	ucred := &ucrednet{uid: 42, pid: 100, socket: dirs.SnapSocket}
+	c.Check(ac.checkAccess(nil, ucred, nil), Equals, accessForbidden)
+
+	// Access allowed from other sockets, or without peer credentials
+	ucred.socket = dirs.SnapdSocket
+	c.Check(ac.checkAccess(nil, ucred, nil), Equals, accessOK)
+	c.Check(ac.checkAccess(nil, nil, nil), Equals, accessOK)
+}
+
+func (s *accessSuite) TestAuthenticatedAccess(c *C) {
+	var ac accessChecker = authenticatedAccess{}
+
+	req := httptest.NewRequest("GET", "/", nil)
+	user := &auth.UserState{}
+
+	// authenticatedAccess denies access from snapd-snap.socket
+	ucred := &ucrednet{uid: 0, pid: 100, socket: dirs.SnapSocket}
+	c.Check(ac.checkAccess(req, ucred, nil), Equals, accessForbidden)
+	c.Check(ac.checkAccess(req, ucred, user), Equals, accessForbidden)
+
+	// With macaroon auth, a normal user is granted access
+	ucred = &ucrednet{uid: 42, pid: 100}
+	c.Check(ac.checkAccess(req, ucred, user), Equals, accessOK)
+
+	// Macaroon access is also possible without peer credentials
+	c.Check(ac.checkAccess(req, nil, user), Equals, accessOK)
```

Comment:
Can you give an example of such access? Would this be the future http-only access or is that something that we may expect to see now?

Reply:
This is the future case of API over TCP, yeah.  As discussed earlier, I agree that it makes sense to have this blocked by default so that we will need to actively decide how authentication should be handled at that point.